### PR TITLE
Update config and run Detekt during the build

### PR DESCRIPTION
**Pull request by yevhenii-nadtochii**
This changeset does the following:

1. Updates `config`.
2. Includes Detekt code analysis into the build and addresses its single finding.
3. Adds in-place `pull` script.
4. Utilizes `Spine` dependency object.
5. Drops unused `travis-warnings` Gradle script.

The version is bumped to `2.0.0-SNAPSHOT.91`.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -104,34 +104,44 @@ all modules and does not describe the project structure per-subproject.
...
   <dependency>
     <groupId>com.puppycrawl.tools</groupId>
     <artifactId>checkstyle</artifactId>
-    <version>10.1</version>
+    <version>10.3.4</version>
+  </dependency>
+  <dependency>
+    <groupId>io.gitlab.arturbosch.detekt</groupId>
+    <artifactId>detekt-cli</artifactId>
+    <version>1.21.0</version>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:139]: ignore

File: pom.xml [hunk 2]
```diff
@@ -104,34 +104,44 @@ all modules and does not describe the project structure per-subproject.
...
     <artifactId>error_prone_annotations</artifactId>
-    <version>2.13.1</version>
+    <version>2.16</version>
     <scope>provided</scope>
   </dependency>
   <dependency>
     <groupId>com.google.errorprone</groupId>
     <artifactId>error_prone_core</artifactId>
-    <version>2.13.1</version>
+    <version>2.16</version>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:113]: ignore

File: pom.xml [hunk 3]
```diff
@@ -142,42 +152,47 @@ all modules and does not describe the project structure per-subproject.
...
     <groupId>org.jetbrains.dokka</groupId>
     <artifactId>dokka-core</artifactId>
-    <version>1.6.20</version>
+    <version>1.7.20</version>
   </dependency>
   <dependency>
     <groupId>org.jetbrains.dokka</groupId>
     <artifactId>javadoc-plugin</artifactId>
-    <version>1.6.20</version>
+    <version>1.7.20</version>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:180]: ignore

File: pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:113]: ignore

**Comment by yevhenii-nadtochii** [comment]
@alexander-yevsyukov @armiol Please take a look.

**Review by armiol — APPROVED** [review]

**Review by alexander-yevsyukov — APPROVED** [review]
LGTM

**Pull request merged**